### PR TITLE
feat: unstaking

### PR DESCRIPTION
**Pull request by 0xsambugs**
We are now adding a way so that the owner can unstake some tokens

### CODE REVIEW

File: src/euler/staking/StakeableEulerERC4626.sol [hunk 1]
```diff
@@ -92,15 +92,20 @@ contract StakeableEulerERC4626 is EulerERC4626, Owned {
...
+    function unstake(uint256 amount) external onlyOwner {
+        stakingRewards.withdraw(amount);
+    }
+
     /// -----------------------------------------------------------------------
     /// Internal functions
     /// -----------------------------------------------------------------------
 
     function _getStakedBalance() internal view returns (uint256) {
-        IStakingRewards _stakingRewards = stakingRewards;
```
Comment (0xsambugs) [inline src/euler/staking/StakeableEulerERC4626.sol]: This was just a rename for consistency

File: src/test/euler/staking/StakeableEulerERC4626.t.sol [hunk 2]
```diff
@@ -79,24 +79,53 @@ contract StakeableEulerERC4626Test is Test {
...
+        _stake(staked);
 
         assertEq(eToken.balanceOf(address(vault)), deposited - staked);
         assertEq(eToken.balanceOf(address(stakingRewards)), staked);
         assertEq(stakingRewards.balanceOf(address(vault)), staked);
     }
 
+    function testFailNotOwnerUnstaking() public {
+        vault.unstake(1000);
+    }
```
Comment (0xged) [inline src/test/euler/staking/StakeableEulerERC4626.t.sol:96]: How do we know is testing `NotOwner` and not another error? 😢

File: src/test/euler/staking/StakeableEulerERC4626.t.sol [hunk 3]
```diff
@@ -106,4 +135,9 @@ contract StakeableEulerERC4626Test is Test {
         rewardsDistribution.setDistribution(index, IRewardsDistribution.DistributionData(address(stakingRewards_), 100000));
     }
 
+    modifier withStakingContract {
+        _setStakingContract();
+        _;
+    }
```
Comment (0xged) [inline src/test/euler/staking/StakeableEulerERC4626.t.sol:141]: Nice idea.

**Review by 0xged — APPROVED** [review]

File: src/test/euler/staking/StakeableEulerERC4626.t.sol (hunk 2 above)
Comment (0xsambugs) [reply src/test/euler/staking/StakeableEulerERC4626.t.sol:96]: Added it in https://github.com/Mean-Finance/yield-daddy/pull/13

**Pull request merged**